kotlin-sdk-test/src/commonTest/kotlin/io/modelcontextprotocol/kotlin/sdk/integration/WebSocketIntegrationTest.kt

```diff
@@ -0,0 +1,207 @@
+package io.modelcontextprotocol.kotlin.sdk.integration
+
+import io.ktor.client.HttpClient
+import io.ktor.server.application.ApplicationStopped
+import io.ktor.server.application.install
+import io.ktor.server.cio.CIOApplicationEngine
+import io.ktor.server.engine.EmbeddedServer
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.routing.routing
+import io.modelcontextprotocol.kotlin.sdk.GetPromptRequest
+import io.modelcontextprotocol.kotlin.sdk.GetPromptResult
+import io.modelcontextprotocol.kotlin.sdk.Implementation
+import io.modelcontextprotocol.kotlin.sdk.PromptArgument
+import io.modelcontextprotocol.kotlin.sdk.PromptMessage
+import io.modelcontextprotocol.kotlin.sdk.Role
+import io.modelcontextprotocol.kotlin.sdk.ServerCapabilities
+import io.modelcontextprotocol.kotlin.sdk.TextContent
+import io.modelcontextprotocol.kotlin.sdk.client.Client
+import io.modelcontextprotocol.kotlin.sdk.client.mcpWebSocketTransport
+import io.modelcontextprotocol.kotlin.sdk.server.Server
+import io.modelcontextprotocol.kotlin.sdk.server.ServerOptions
+import io.modelcontextprotocol.kotlin.sdk.server.mcpWebSocket
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import kotlinx.coroutines.withTimeout
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import io.ktor.client.engine.cio.CIO as ClientCIO
+import io.ktor.client.plugins.websocket.WebSockets as ClientWebSocket
+import io.ktor.server.cio.CIO as ServerCIO
+import io.ktor.server.websocket.WebSockets as ServerWebSockets
+
+class WebSocketIntegrationTest {
+
+    @Test
+    fun `client should be able to connect to websocket server 2`() = runTest {
+        var server: EmbeddedServer<CIOApplicationEngine, CIOApplicationEngine.Configuration>? = null
+        var client: Client? = null
+
+        try {
+            withContext(Dispatchers.Default) {
+                withTimeout(1000) {
+                    server = initServer()
+                    val port = server.engine.resolvedConnectors().first().port
+                    client = initClient(serverPort = port)
```

Comment:
let's actually verify something

```suggestion
                    client = initClient(serverPort = port)
                    
                    val serverVersion = client.serverVersion
                    assertEquals("websocket-server", serverVersion?.name)
                    assertEquals("1.0.0", serverVersion?.version)
```